float64 add_offset
---
float64 total_offset
